packages/expo-modules-core/ios/Swift/ExpoBridgeModule.swift

```diff
@@ -41,6 +41,30 @@ public final class ExpoBridgeModule: NSObject, RCTBridgeModule {
   public var bridge: RCTBridge! {
     didSet {
       appContext.reactBridge = bridge
+
+      if bridge.responds(to: Selector(("runtime"))) {
+        // Getting the `runtime` on a different thread than JS is considered to be dangerous.
+        // However, we just checking if it exists. We don't do anything with it.
+        let result = bridge.perform(Selector(("runtime")))
+        if result == nil {
+          // When exporting expo modules using `extraModulesForBridge` (e.g. in Expo Go),
+          // the runtime won't be initiated before the bidge didSet method is called.
+          // Therefore, we need to wait for the main thread to complete its initialization by dispatching on it first.
+          bridge.dispatchBlock({ [weak self] in
+            guard let self = self, let bridge = self.appContext.reactBridge else {
+              return
+            }
+            bridge.dispatchBlock({ [weak self] in
+              guard let self = self, let bridge = self.appContext.reactBridge else {
+                return
+              }
+              self.appContext.runtime = EXJavaScriptRuntimeManager.runtime(fromBridge: bridge)
+            }, queue: RCTJSThread)
+          }, queue: DispatchQueue.main)
+          return
+        }
+      }
```

Comment:
Could you extract the contents of this `if` to a separate function? Also it seems that dispatchBlock on the JS thread is used twice with the same code, so that could be a separate function as well.